app/controllers/caseload_controller.rb

```diff
@@ -22,7 +22,36 @@ def new
 private
 
   def sort_allocations(allocations)
-    sort_collection(allocations, default_sort: :last_name)
+    field, direction = sort_params(:last_name)
+
+    if field == :cell_location
+      cell_location_sort(allocations, direction)
+    else
+      sort_collection(allocations, default_sort: :last_name)
+    end
+  end
+
+  def cell_location_sort(allocations, direction)
+    allocations = allocations.sort do |a, b|
+      if a.latest_movement_date.nil? && b.latest_movement_date.nil?
+        if a.cell_location.present? && b.cell_location.present?
+          a.cell_location <=> b.cell_location
+        elsif a.cell_location.nil?
+          -1
+        else
+          1
+        end
+      elsif a.latest_movement_date.nil? && b.latest_movement_date.present?
+        1
+      elsif a.latest_movement_date.present? && b.latest_movement_date.nil?
+        -1
+      else
+        a.latest_movement_date <=> b.latest_movement_date
+      end
+    end
```

Comment:
As we said: this method feels a bit confusing if you don't know what it's about.

It might be worth adding a comment to explain (at a high level) what it does – just to help anyone who stumbles across it in the future.